packages/blueprint/package.yaml

```
@@ -0,0 +1,28 @@
+---
+name: blueprint_ls
+description: Language server for Blueprint, a markup language and compiler for GTK 4 user interfaces.
+homepage: https://gitlab.gnome.org/jwestman/blueprint-compiler
+licenses:
+  - LGPL-3.0-or-later
+languages:
+  - Blueprint
+categories:
+  - LSP
+
+source:
+  id: pkg:generic/blueprint?vcs_url=git%2Bhttps://gitlab.gnome.org/jwestman/blueprint-compiler%402a39a16391122af2f3d812e478c1c1398c98b972
```

The mason.nvim client would currently not be able to parse this PURL.

The purl spec includes a `gitlab` pkg type, which I think would be more fitting here. However, mason.nvim currently doesn't support this package type, so that would have to be added first.

```suggestion
  id: pkg:gitlab/jwestman/blueprint-compiler@402a39a16391122af2f3d812e478c1c1398c98b972
```

---

Oh it's a self-hosted instance, that complicates things. The generic type might be better.

---

I saw the type, but was not sure if it would work since the repository is not hosted on the gitlab.com instance. Will it still work?
What is required for mason.nvim to support the package type?